### PR TITLE
ws-manager-bridge: Print audit logs when client header is set

### PR DESCRIPTION
**Pull request by princerachit**
## Description

Print the operation and the client info.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7858

## How to test

This PR can be tested via the changes of https://github.com/gitpod-io/gitpod/pull/7976.

The logs were visible in the `ws-manager-bridge` when `gpctl clusters list` was run using binary generated from above PR.

```json
{"component":"ws-manager-bridge","severity":"INFO","time":"2022-02-04T05:50:13.917Z","message":"requested clusters.list","payload":{"clientIP":"127.0.0.1:57924","clientName":"gpctl:gitpod@gitpodio-gitpod-36z8o2s861y","userAgent":"grpc-go/1.39.1"}}
```

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **alextugarev** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @alextugarev` in a comment when ready.

Associated issue: *#7858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ws-manager-bridge/src/cluster-service-server.ts
Comment (princerachit) [inline components/ws-manager-bridge/src/cluster-service-server.ts:159]: I tam not sure if we should add what updates were made here and in other methods.

**Comment by csweichel** [comment]
A consequence of this design is that if someone wanted to circumvent the audit log they just don't need to send their name. We should always log, even if we didn't get the name. In addition, see if you can get connection information, e.g. the source IP address.

Also, why not register a single interceptor on the server which produces the log rather than adding it to the concrete implementation of the functions?
Lastly, please prefer structured logging over interpolating the log message.

**Comment by princerachit** [comment]
Thanks @csweichel for the review, I could not find server [grpc interceptors for the grpc-js](https://github.com/gitpod-io/gitpod/issues/7858#issuecomment-1027682390) library. Are you talking about something else?

**Comment by csweichel** [comment]
That is a fair point and an unexpected omission :/

File: components/ws-manager-bridge/src/cluster-service-server.ts
Comment (geropl) [inline components/ws-manager-bridge/src/cluster-service-server.ts:375]: If you return a shape ` { clientName, userAgent }` that's in line with all other usages of `log.info`, and we make sure we can properly filter it in GCloud, for instance.

File: components/ws-manager-bridge/src/cluster-service-server.ts
Comment (geropl) [inline components/ws-manager-bridge/src/cluster-service-server.ts:213]: the order of the arguments should be reversed: `log.info("requested clusters.deregister", getUserAgentAndClientName(call))`

**Review by geropl — CHANGES_REQUESTED** [review]

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#7858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ws-manager-bridge/src/cluster-service-server.ts
Comment (princerachit) [reply components/ws-manager-bridge/src/cluster-service-server.ts]: done

File: components/ws-manager-bridge/src/cluster-service-server.ts
Comment (sagor999) [inline components/ws-manager-bridge/src/cluster-service-server.ts:375]: is it possible to also log ip address here? for example, I assume if someone is using old gpctl, it will not send `client-name`, and maybe not even `user-agent`. it would be good to at least log some sort of breadcrumb.

Comment (princerachit) [reply components/ws-manager-bridge/src/cluster-service-server.ts]: I initially refrained from adding peer information because it was returning `127.0.0.1`. However, I have added it now.

File: components/ws-manager-bridge/src/cluster-service-server.ts
Comment (geropl) [inline components/ws-manager-bridge/src/cluster-service-server.ts:376]: :+1:

**Comment by geropl** [comment]
Testing...

**Review by geropl — APPROVED** [review]
Code looks good, tested and works :+1:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: b6a841f241abc6bf9550a1b422df5d0212c03267</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7982#pullrequestreview-872761811" title="Approved">geropl</a>*

Associated issue: *#7858*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
